packages/core/src/queries/application-sign-in-experience.ts

```diff
@@ -0,0 +1,37 @@
+import { ApplicationSignInExperiences, type ApplicationSignInExperience } from '@logto/schemas';
+import { convertToIdentifiers } from '@logto/shared';
+import { sql, type CommonQueryMethods } from 'slonik';
+
+import { buildInsertIntoWithPool } from '#src/database/insert-into.js';
+import { buildUpdateWhereWithPool } from '#src/database/update-where.js';
+
+const createApplicationSignInExperienceQueries = (pool: CommonQueryMethods) => {
+  const insert = buildInsertIntoWithPool(pool)(ApplicationSignInExperiences, {
+    returning: true,
+  });
+
+  const safeFindSignInExperienceByApplicationId = async (applicationId: string) => {
```

Comment:
The "safe" means it won't throw 404 error?

Reply:
yeap returns a nullable result.